CommonFindPackage.cmake

```diff
@@ -45,54 +45,46 @@ option(COMMON_FIND_PACKAGE_QUIET "Use QUIET for common_find_package command" ON)
 
 macro(common_find_package Package_Name)
   string(TOUPPER ${Package_Name} PACKAGE_NAME)
-  set(__args ${ARGN}) # ARGN is not a list. make one.
 
-  # SYSTEM specified?
-  set(__is_system_package)
-  if(__args)
-    string(REGEX MATCH "SYSTEM?" __is_system_package ${__args})
-    list(REMOVE_ITEM __args "SYSTEM")
-  endif()
+  # Parse macro arguments
+  set(__options QUIET REQUIRED SYSTEM)
+  set(__oneValueArgs MODULE)
+  set(__multiValueArgs)
+  cmake_parse_arguments(__pkg "${__options}" "${__oneValueArgs}" "${__multiValueArgs}" ${ARGN})
 
   # Boost is always SYSTEM
   if(${Package_Name} STREQUAL "Boost")
-    set(__is_system_package ON)
+    set(__pkg_SYSTEM TRUE)
```

Comment:
__pkg_SYSTEM is not used. There must be a __is_system_package at the end of the function which does include_directories(BEFORE SYSTEM ${__include_dirs}